### PR TITLE
Intial commit for policy update and viewbox support

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
1) Added option to support policy updation.
2) Storage domain creation, updation and deletion support added.

### CODE REVIEW

File: playbooks/storage_domain_workflow/create_storage_domain.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  tasks:
+    - name: Create view box in the cohesity server.
+      cohesity_storage_domain:
+        cluster: "{{ cohesity_server }}"
+        username: "{{ cohesity_username }}"
+        password: "{{ cohesity_password }}"
+        validate_certs: "{{ cohesity_validate_certs }}"
+        name: 'Ansible'
+        state: 'present'
+        cluster_partition_id: 3
```
Comment (pyashish) [inline playbooks/storage_domain_workflow/create_storage_domain.yaml:16]: How does one get this ID , any API / documentation ?

File: plugins/modules/cohesity_storage_domain.py [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+__metaclass__ = type
+
+# GNU General Public License v3.0+ (see https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+DOCUMENTATION = r"""
+---
+author: "Naveena (@naveena-maplelabs)"
+description:
+  - "Ansible Module used to register or remove the Cohesity Protection Sources to/from a Cohesity Cluster."
```
Comment (pyashish) [inline plugins/modules/cohesity_storage_domain.py:16]: This is incorrect ?

File: plugins/modules/cohesity_storage_domain.py [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+  - "state action will be applied."
+module: cohesity_storage_domain
+options:
+  cluster:
+    aliases:
+      - cohesity_server
+    description:
+      - "IP or FQDN for the Cohesity Cluster"
+    type: str
+  cohesity_admin:
```
Comment (pyashish) [inline plugins/modules/cohesity_storage_domain.py:27]: This should just be cohesity_user ?

File: plugins/modules/cohesity_storage_domain.py [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+  cohesity_password:
+    aliases:
+      - password
+      - admin_pass
+    description:
+      - "Password belonging to the selected Username. This parameter will not be logged."
+    type: str
+  ad_domain_name:
+    description:
+      - "Specifies an active directory domain that this view box is mapped to."
```
Comment (pyashish) [inline plugins/modules/cohesity_storage_domain.py:47]: Remove the term view box from everywhere.

File: plugins/modules/cohesity_storage_domain.py [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+    password: password
+    name: StorageDomain
+    state: absent
+"""
+
+RETURN = """
+"""
+import json
+
+from ansible.module_utils.basic import AnsibleModule
```
Comment (pyashish) [inline plugins/modules/cohesity_storage_domain.py:115]: Have we implemented check mode ? if not please do

File: plugins/modules/cohesity_storage_domain.py [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+
+    except APIException as ex:
+        raise__cohesity_exception__handler(
+            str(json.loads(ex.context.response.raw_body)), module
+        )
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def delete_domain(module, domain_id):
```
Comment (pyashish) [inline plugins/modules/cohesity_storage_domain.py:309]: *delete_storage_domain
, here and below doc string.

**Review by pyashish — CHANGES_REQUESTED** [review]

File: plugins/modules/cohesity_storage_domain.py (hunk 2 above)
Comment (naveena-maplelabs) [reply plugins/modules/cohesity_storage_domain.py:16]: Modified this

File: plugins/modules/cohesity_storage_domain.py (hunk 3 above)
Comment (naveena-maplelabs) [reply plugins/modules/cohesity_storage_domain.py:27]: We are using the same format in all the other modules. Can we change in all the modules?

File: playbooks/storage_domain_workflow/create_storage_domain.yaml (hunk 1 above)
Comment (naveena-maplelabs) [reply playbooks/storage_domain_workflow/create_storage_domain.yaml:16]: cluster_partition_name is also supported.

File: playbooks/storage_domain_workflow/create_storage_domain.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  tasks:
+    - name: Create view box in the cohesity server.
+      cohesity_storage_domain:
+        cluster: "{{ cohesity_server }}"
+        username: "{{ cohesity_username }}"
+        password: "{{ cohesity_password }}"
+        validate_certs: "{{ cohesity_validate_certs }}"
+        name: 'Ansible'
+        state: 'present'
+        cluster_partition_name: "DefaultPartition"
```
Comment (pyashish) [inline playbooks/storage_domain_workflow/create_storage_domain.yaml:16]: Question still remains, how does one get to know the partition name, if its is constant, can we just hard code it ?

**Pull request merged**